Dockerfile

```diff
@@ -49,29 +50,36 @@ ENV PATH /usr/local/bin:$PATH
 RUN cd /usr/local/bin && \
   ln -s $(which python3) python
 
-# install python dependencies
-RUN pip install -r requirements.txt && \
-  useradd -lm kivy
-
 RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
     locale-gen
 ENV LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
 
+# Install Android SDK
+ENV ANDROID_HOME=/opt/android
+ENV ANDROIDSDK=$ANDROID_HOME/sdk
+ENV ANDROIDNDK=$ANDROIDSDK/ndk-bundle
+COPY Makefile /tmp/
+RUN make -C /tmp setup SDK=$ANDROIDSDK && \
+  rm -f /tmp/Makefile
+
+# install python dependencies
+COPY requirements.txt /tmp/
+RUN pip install -r /tmp/requirements.txt && \
+  rm -f /tmp/requirements.txt && \
+  useradd -lm kivy
+
 USER kivy:kivy
 WORKDIR /home/kivy
 
-# make sure the build is from scratch on a per-architecture basis
-ARG ARCH=$ARCH
```

Comment:
I think the `ARCH` environment variable definitely does have use. All I'm saying is that in the Dockerfile it doesn't actually do anything. If you wanted to make it work there, you'd have to actually use `--build-arg` somewhere. docker build doesn't use environment variables. I tried both of the following and neither did anything when I had ARCH set as an environment variable:

```
FROM debian:stable-slim
ARG ARCH=${ARCH}
RUN echo ARCH=$ARCH
```

```
FROM debian:stable-slim
ENV ARCH=${ARCH}
ARG ARCH=${ARCH}
RUN echo ARCH=$ARCH
```

Only when I used `--build-arg` or changed to `ENV ARCH=something` did it actually echo anything.

Reply:
One thing to flag is that once we upgrade to the latest version of Python-for-Android, it now has multi-arch support, so we may not need to do more than one build, if I am understanding it correctly.

Reply:
I skimmed past the earlier comment about fear of using the same image for 64 and 32 bit builds. I actually think it's fine to use the same image for both. Nothing in the image depends on the target architecture. I also think it would be fine to share the cache between architectures. p4a separates the build outputs by target architecture. But maybe I missed something there.

And like @rtibbles says, with the current version of p4a you can do a multi-arch build.

Reply:
Except that they haven't enabled it for the webview bootstrap that we use yet, so we'll probably have to do that port on our fork, and contribute it back upstream.